core/server/DatabaseAPI.cpp

```diff
@@ -964,65 +954,63 @@ void DatabaseAPI::recordCreateBatch(
 void DatabaseAPI::recordUpdate(const Auth::RecordUpdateRequest &a_request,
                                Auth::RecordDataReply &a_reply,
                                libjson::Value &result, LogContext log_context) {
-  string body = "{\"id\":\"" + a_request.id() + "\"";
+  nlohmann::json payload;
+  payload["id"] = a_request.id();
   if (a_request.has_title())
-    body += ",\"title\":\"" + escapeJSON(a_request.title()) + "\"";
+    payload["title"] = escapeJSON(a_request.title());
   if (a_request.has_desc())
-    body += ",\"desc\":\"" + escapeJSON(a_request.desc()) + "\"";
+    payload["desc"] = escapeJSON(a_request.desc());
   if (a_request.has_alias())
-    body += ",\"alias\":\"" + a_request.alias() + "\"";
+    payload["alias"] = a_request.alias();
 
   if (a_request.has_tags_clear() && a_request.tags_clear()) {
-    body += ",\"tags_clear\":true";
+    payload["tags_clear"] = a_request.tags_clear();
   } else if (a_request.tags_size()) {
-    body += ",\"tags\":[";
+    nlohmann::json tags = nlohmann::json::array();
     for (int i = 0; i < a_request.tags_size(); i++) {
-      if (i)
-        body += ",";
-      body += "\"" + a_request.tags(i) + "\"";
+      tags.push_back(a_request.tags(i));

```

Comment:
I prefer always wrapping if statement with {} to prevent accidental newlines breaking things.

Reply:
Would again recommed using range based for

```
for (const auto& tag : a_request.tags()) {
    tags.push_back(tag);
}
```